lib/battle/DamageCalculator.cpp

```diff
@@ -213,18 +201,9 @@ double DamageCalculator::getAttackBlessFactor() const
 
 double DamageCalculator::getAttackOffenseArcheryFactor() const
 {
-	if(info.shooting)
-	{
-		const std::string cachingStrArchery = "type_SECONDARY_SKILL_PREMYs_ARCHERY";
-		static const auto selectorArchery = Selector::typeSubtype(Bonus::SECONDARY_SKILL_PREMY, SecondarySkill::ARCHERY);
-		return info.attacker->valOfBonuses(selectorArchery, cachingStrArchery) / 100.0;
-	}
-	else
-	{
-		const std::string cachingStrOffence = "type_SECONDARY_SKILL_PREMYs_OFFENCE";
-		static const auto selectorOffence = Selector::typeSubtype(Bonus::SECONDARY_SKILL_PREMY, SecondarySkill::OFFENCE);
-		return info.attacker->valOfBonuses(selectorOffence, cachingStrOffence) / 100.0;
-	}
+	const auto cachingStr = "type_PERCENTAGE_DAMAGE_BOOSTs_" + std::to_string(info.shooting);
+	static const auto selector = Selector::typeSubtype(Bonus::PERCENTAGE_DAMAGE_BOOST, info.shooting);
```

Comment:
"static const" variable is initialized with "info.shooting" which may be true or false

Reply:
@IvanSavenko is okay to remove static here? will it hurt performance much?

Reply:
@IvanSavenko returned to previous form with 2 static const selectors.